Apps/Sandcastle/gallery/Multiple Synced Views.html

```diff
@@ -0,0 +1,106 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no">
+    <meta name="description" content="Multiple views synced across time and space.">
+    <meta name="cesium-sandcastle-labels" content="Beginner, Showcases">
+    <title>Cesium Demo</title>
+    <script type="text/javascript" src="../Sandcastle-header.js"></script>
+    <script type="text/javascript" src="../../../ThirdParty/requirejs-2.1.20/require.js"></script>
+    <script type="text/javascript">
+    require.config({
+        baseUrl : '../../../Source',
+        waitSeconds : 60
+    });
+    </script>
+</head>
+<body class="sandcastle-loading" data-sandcastle-bucket="bucket-requirejs.html">
+<style>
+    @import url(../templates/bucket.css);
+    #cesiumContainer {
+      display: flex;
+      width: 100%;
+      height: 100%;
+    }
+    #view3D {
+      display: inline-block;
+      width: 100%;
+    }
+    #view2D {
+      display: inline-block;
+      width: 100%;
+    }
+</style>
+<div id="cesiumContainer" class="fullSize">
+    <div id="view3D"></div>
+    <div id="view2D"></div>
+</div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar"></div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+// We want our two views to be synced across time, so we create
+// a shared clock object that both views share
+var clockViewModel = new Cesium.ClockViewModel(new Cesium.Clock());
+var options3D = {
+    homeButton:false,
+    fullscreenButton:false,
+    sceneModePicker:false,
+    clockViewModel: clockViewModel,
+};
+var options2D = {
+    homeButton:false,
+    fullscreenButton:false,
+    sceneModePicker:false,
+    clockViewModel: clockViewModel,
+    infoBox:false,
+    geocoder:false,
+    sceneMode: Cesium.SceneMode.SCENE2D,
+    navigationHelpButton:false,
+    animation:false
+};
+// We create two viewers, a 2D and a 3D one
+// The CSS is set up to place them side by side
+var view3D = new Cesium.Viewer('view3D', options3D);
+var view2D = new Cesium.Viewer('view2D', options2D);
+
+function sync2DView(){
+    // The center of the view is the point that the 3D camera is focusing on
+    var viewCenter = new Cesium.Cartesian2(Math.floor(view3D.canvas.clientWidth / 2), Math.floor(view3D.canvas.clientHeight / 2));
+    // Given the pixel in the center, get the world position
+    var worldPosition = view3D.scene.camera.pickEllipsoid(viewCenter);
+    // Get the distance between the world position of the point the camera is focusing on, and the camera's world position
+    var distance = Cesium.Cartesian3.distance(worldPosition, view3D.scene.camera.positionWC);
```

Comment:
When I tilt the camera view and the center of the camera is not on the ellipsoid, I get a developer error.

You'll probably want to check if world position is defined, and if not, just assume a base value for `distance`, or use a previous value.

Reply:
Also to check if an object is defined, we use [`Cesium.defined`](https://cesiumjs.org/Cesium/Build/Documentation/defined.html).